modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/ChartStateContext.js

```diff
@@ -49,25 +49,46 @@ export const ChartStateContextProvider = ({
 	);
 };
 
-export const useChartState = () => {
-	const [state, dispatch] = useContext(ChartStateContext);
+export const useAddDataSetItems = () => {
+	const [, dispatch] = useContext(ChartStateContext);
 	const {validAnalyticsConnection} = useContext(ConnectionContext);
 
-	const actionsRef = useRef({
-		addDataSetItems: (payload) =>
-			dispatch({
-				payload,
-				type: ADD_DATA_SET_ITEMS,
-				validAnalyticsConnection,
-			}),
-		changeTimeSpanKey: (payload) =>
-			dispatch({payload, type: CHANGE_TIME_SPAN_KEY}),
-		nextTimeSpan: () => dispatch({type: NEXT_TIME_SPAN}),
-		previousTimeSpan: () => dispatch({type: PREV_TIME_SPAN}),
-		setLoading: () => dispatch({type: SET_LOADING}),
-	});
+	return (payload) =>
```

Comment:
If you're using these functions inside useEffect calls you should wrap them with a useCallback, otherwise the useEffect will be triggered on every component call.

Reply:
Ok, I'll add useCallback in those that I used inside useEffect calls 👍  Thanks!

Reply:
In this case I will change `addDataSetItems` and `setLoading` (`Chart` component).